scripts/core/editor3/components/embeds/EmbedBlock.tsx

```diff
@@ -59,14 +63,22 @@ export class EmbedBlockComponent extends React.Component<IProps> {
 
         tree.find('script').each((i, s) => {
             if (s.hasAttribute('src')) {
-                let url = s.getAttribute('src');
-                const async = s.hasAttribute('async');
+                const newScript = document.createElement('script');
+                const oldKeys = Object.keys(window);
 
-                if (url.startsWith('http')) {
-                    url = url.substring(url.indexOf(':') + 1);
-                }
+                newScript.setAttribute('src', s.getAttribute('src'));
+                newScript.setAttribute('async', s.hasAttribute('async') ? 'async' : '');
+                document.body.appendChild(newScript);
+                newScript.onload = () => {
+                    const newKeys = Object.keys(window);
+
+                    // keep track of new global objects created after the script was loaded
+                    this.globalKeys = newKeys.filter((k) => !oldKeys.includes(k) && isNaN(parseInt(k, 10)));
```

Comment:
I'm not following what's being done with `globalKeys`, but in any case it's not reliable because it might pick up keys that other parts of the app are setting and would interfere. We don't use globals much so impact isn't significant, but still would be good to avoid. What's the purpose of it?

Reply:
when you load eg. twitter embed script, it will first time create some `window.twitter` object and handle all objects with some `data-twitter` param. but if you load the script for the second time it will detect that `window.twitter` already exists and won't do anything. so removing such global object will make it initialize on next script load again.